RULES/cc/cc.go

```diff
@@ -99,26 +97,56 @@ type Library struct {
 	Shared        bool
 	AlwaysLink    bool
 	Toolchain     Toolchain
+
+	multipleToolchains bool
+	toolchainMap       map[string]Library
+	baseOut            core.OutPath
+}
+
+func (lib Library) MultipleToolchains() Library {
+	lib.multipleToolchains = true
+	lib.toolchainMap = make(map[string]Library)
+	lib.baseOut = lib.Out
+	return lib
 }
 
 // Build a Library.
 func (lib Library) Build(ctx core.Context) {
 	toolchain := lib.Toolchain
 	if toolchain == nil {
-		toolchain = &DefaultToolchain
+		toolchain = defaultToolchain()
+	}
+
+	if lib.multipleToolchains {
+		if lib.Toolchain == nil {
+			var defaultLib = lib.WithToolchain(ctx, defaultToolchain())
+			ctx.AddBuildStep(core.BuildStep{
```

Comment:
There is actually a copy rule, in case you want to use that

Reply:
yeah sorry I changed that while you were reviewing.